### PR TITLE
Saas 1569 rework auth

### PR DESCRIPTION
**Pull request by yaroslav-codefresh**

### CODE REVIEW

File: lib/auth/contexts/APIKeyContext.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class APIKeyContext extends Context {
 
     async validate() {
         try {
-            return await whoami(this);
+            return await whoami(this, true);
```
Comment (olegsu) [inline lib/auth/contexts/APIKeyContext.js]: do not `return await` just `return`

Comment (yaroslav-codefresh) [reply lib/auth/contexts/APIKeyContext.js]: it's for error handling

File: lib/auth/contexts/whoami.js [hunk 2]
```diff
@@ -3,19 +3,21 @@ const debug = require('debug')('codefresh:sdk:auth:whoami');
...
 const { Http } = require('../../../helpers/http');
 
-const http = Http({
-    timeout: 2000,
-    retryStrategy: () => false,
-});
+const http = Http();
 
-const whoami = async (context) => {
+const whoami = async (context, fullUser = false) => {
```
Comment (olegsu) [inline lib/auth/contexts/whoami.js]: return always the user, extract the account in other func

File: lib/auth/contexts/whoami.js [hunk 3]
```diff
@@ -3,19 +3,21 @@ const debug = require('debug')('codefresh:sdk:auth:whoami');
 
 const { Http } = require('../../../helpers/http');
 
-const http = Http({
-    timeout: 2000,
-    retryStrategy: () => false,
-});
+const http = Http();
```
Comment (olegsu) [inline lib/auth/contexts/whoami.js]: move this into the func- try to not have global variables

Comment (yaroslav-codefresh) [reply lib/auth/contexts/whoami.js]: it's ok for this place, it needs to be initialized only once. Or anyway move?

**Pull request merged**